__init__.py

```diff
@@ -87,94 +218,68 @@ def handle_tell_more(self, message):
-
-            # First step is to get wiki article titles.  This comes back
-            # as a list.  I.e. "beans" returns ['beans',
-            #     'Beans, Beans the Music Fruit', 'Phaseolus vulgaris',
-            #     'Baked beans', 'Navy beans']
-            results = wiki.search(search, 5)
-            if len(results) == 0:
-                self.speak_dialog("no entry found")
-                return
-
-            # Now request the summary for the first (best) match.  Wikipedia
-            # writes in inverted-pyramid style, so the first sentence is the
-            # most important, the second less important, etc.  Two sentences
-            # is all we ever need.
-            lines = 2
-            summary = wiki.summary(results[0], lines,
-                                   auto_suggest=auto_suggest)
-
-            if "==" in summary or len(summary) > 250:
-                # We hit the end of the article summary or hit a really long
-                # one.  Reduce to first line.
-                lines = 1
-                summary = wiki.summary(results[0], lines,
-                                       auto_suggest=auto_suggest)
-
-            # Now clean up the text and for speaking.  Remove words between
-            # parenthesis and brackets.  Wikipedia often includes birthdates
-            # in the article title, which breaks up the text badly.
-            summary = re.sub(r'\([^)]*\)|/[^/]*/', '', summary)
-
-            # Send to generate displays
-            self.gui.clear()
-            pagetext = wiki.page(results[0], auto_suggest=auto_suggest)
-            self.gui['summary'] = summary
-            self.gui['imgLink'] = wiki_image(pagetext)
-            self.gui.show_page("WikipediaDelegate.qml", override_idle=60)
-
-            # Remember context and speak results
-            self.set_context("wiki_article", results[0])
-            self.set_context("spoken_lines", str(lines))
-            self.speak(summary)
-            self.results = results
-
-        except wiki.exceptions.DisambiguationError as e:
-            # Test:  "tell me about john"
-            options = e.options[:5]
-
-            option_list = (", ".join(options[:-1]) + " " +
-                           self.translate("or") + " " + options[-1])
-            choice = self.get_response('disambiguate',
-                                       data={"options": option_list})
-            if choice:
-                self._lookup(choice, auto_suggest=auto_suggest)
+        lang_code = self.translate_namedvalues("wikipedia_lang")['code']
+
+        def lookup(auto_suggest):
+            try:
+                return wiki_lookup(search, lang_code, auto_suggest)
+            except wiki.PageError:
+                return None
+            except Exception as e:
+                self.log.error("Error: {0}".format(e))
+                return None
+
+        with ThreadPoolExecutor() as pool:
+            results = list(pool.map(lookup, (True, False)))
```

Comment:
If there will always be two results is it worth clarifying what they are?
eg something like `result_using_auto_suggest, result_without_auto_suggest = ...`

That might help make the return values below more understandable? eg instead of `results[0]`

Reply:
good suggestion, will add that